language/src/ceylon/language/ArraySequence.ceylon

```diff
@@ -3,127 +3,106 @@
     
-    each(void step(Element element)) => array.each(step);
+    each = array.each;
     
-    count(Boolean selecting(Element element))
-            => array.count(selecting);
+    count = array.count;
     
-    every(Boolean selecting(Element element))
-            => array.every(selecting);
+    every = array.every;
     
-    any(Boolean selecting(Element element))
-            => array.any(selecting);
+    any = array.any;
     
-    find(Boolean selecting(Element&Object element))
-            => array.find(selecting);
+    find = array.find;
     
-    findLast(Boolean selecting(Element&Object element))
-            => array.findLast(selecting);
+    findLast = array.findLast;
     
     shared actual 
-    Result|Element reduce<Result>(
-        Result accumulating(Result|Element partial, 
-                            Element element)) {
-        assert (exists result 
-            = array.reduce(accumulating));
+    Result|Element reduce<Result>(accumulating) {
+        Result accumulating(Result|Element partial, Element element);
+        assert(is Element result = array.reduce(accumulating));
         return result;
     }
 
-    shared actual 
-    [Result+] collect<Result>
-            (Result collecting(Element element)) {
-        assert (nonempty sequence 
-            = array.collect(collecting));
+    shared actual [Result+] collect<Result>(collecting) {
+        Result collecting(Element element);
+        assert(nonempty sequence = array.collect(collecting));
         return sequence;
     }
     
-    shared actual 
-    [Element+] sort
-            (Comparison comparing(Element x, Element y)) {
-        assert (nonempty sequence 
-            = array.sort(comparing));
+    shared actual [Element+] sort(comparing) {
+        Comparison comparing(Element x, Element y);
+        assert(nonempty sequence = array.sort(comparing));
         return sequence;
     }
     
-    shared actual 
-    Element[] measure(Integer from, Integer length) {
-        if (from > lastIndex || 
-            length <= 0 || 
-            from + length <= 0) {
+    shared actual
+    ArraySequence<Element>|[] measure(Integer from, Integer length) {
+        if(from > lastIndex || length <= 0 || from + length <= 0) {
             return [];
         }
         else {
-            return ArraySequence(array[from : length]);
+            return ArraySequence(array[from:length]);
         }
     }
     
-    shared actual 
-    Element[] span(Integer from, Integer to) {
-        if (from <= to) {
+    shared actual
+    ArraySequence<Element>|[] span(Integer from, Integer to) {
+        if(from <= to) {
             return 
-                if (to < 0 || from > lastIndex)
-                then [] 
-                else ArraySequence(array[from..to]);
+            if(to < 0 || from > lastIndex) {
+                return [];
+            }
+            else {
+                return ArraySequence(array[from..to]);
+            }
```

Comment:
I think if expressions particularly read better in these scenarios where both the if and else branches each need to `return`.